### PR TITLE
Low hanging fruit regarding limits

### PR DESCRIPTION
**Pull request by TOTBWF**
# Description

This PR fills in a bunch of low hanging fruit from chapter 2 of Borceux, Vol 1.

## Checklist

Before submitting a merge request, please check the items below:

- [x] I've read [the contributing guidelines](https://github.com/plt-amy/1lab/blob/main/CONTRIBUTING.md).
- [x] The imports of new modules have been sorted with `support/sort-imports.hs` (or `nix run --experimental-features nix-command -f . sort-imports`).
- [x] All new code blocks have "agda" as their language.

If your change affects many files without adding substantial content, and
you don't want your name to appear on those pages (for example, treewide
refactorings or reformattings), start the commit message and PR title with `chore:`.

### CODE REVIEW

File: src/Cat/Diagram/Product/Indexed.lagda.md [hunk 1]
```diff
@@ -233,3 +272,72 @@ lying over our isomorphism.
...
+successive product.
+
+```agda
+    πᵃᵇ' : ∀ (ab : Σ A B) → C.Hom ΠᵃΠᵇX (X ab)
+    πᵃᵇ' (a , b) = πᵇ a b C.∘ πᵃ a
+```
+
+The rest of the structure follows a similar pattern.
+
+```
```
Comment (plt-amy) [inline src/Cat/Diagram/Product/Indexed.lagda.md:317]:
```suggestion
```agda
```

think this is the only thing causing the CI failure

**Comment by Lavenza** [comment]
[Pull request preview](https://preview.1lab.dev/410)

File: src/Cat/Diagram/Pullback.lagda.md [hunk 2]
```diff
@@ -152,6 +152,201 @@ module _ {o ℓ} {C : Precategory o ℓ} where
...
+
+```agda
+  is-kernel-pair→epil
+    : ∀ {p1 p2 : Hom P X} {f : Hom X Y}
+    → is-kernel-pair C p1 p2 f
+    → is-epic p1
+```
+
+Recall that a morphism is epic if it has a [[section]]; that is,
+some morphism $u$ such that $p_1 \circ u = id$. We can construct
```
Comment (ncfavier) [inline src/Cat/Diagram/Pullback.lagda.md:196]: `\id`

File: src/Cat/Diagram/Equaliser.lagda.md [hunk 3]
```diff
@@ -106,6 +106,70 @@ module _ {o ℓ} {C : Precategory o ℓ} where
...
+  equaliser+epi→invertible
+    : ∀ {E} {e : Hom E A}
+    → is-equaliser C f g e
+    → is-epic e
+    → is-invertible e
+```
+
+Suppose that $e$ equalises some pair $(f, g) : A \to B$. By definition,
+this means that $f \circ e = g \circ e$; however, $e$ is an epi, so
+$f = g$. This in turn means that $id : A \to A$ can be extended into
```
Comment (ncfavier) [inline src/Cat/Diagram/Equaliser.lagda.md:156]: `\id`

File: src/Cat/Diagram/Pullback.lagda.md [hunk 4]
```diff
@@ -152,6 +152,201 @@ module _ {o ℓ} {C : Precategory o ℓ} where
...
+    has-section→epic $
+    make-section
+      (universal refl)
+      p₂∘universal
+    where open is-pullback is-kp
+```
+-->
+
+If $f$ is a [[monomorphism]], then its kernel pair always exists, and
+is given by $(id, id)$.
```
Comment (ncfavier) [inline src/Cat/Diagram/Pullback.lagda.md:241]: `\id`

File: src/Cat/Diagram/Pullback.lagda.md [hunk 5]
```diff
@@ -152,6 +152,201 @@ module _ {o ℓ} {C : Precategory o ℓ} where
...
+
+```agda
+  monic→id-kernel-pair
+    : ∀ {f : Hom X Y}
+    → is-monic f
+    → is-kernel-pair C id id f
+```
+
+Clearly, the square $f \circ id = f \circ id$ commutes, so the tricky
```
Comment (ncfavier) [inline src/Cat/Diagram/Pullback.lagda.md:250]: `\id`

File: src/Cat/Diagram/Pullback.lagda.md [hunk 6]
```diff
@@ -152,6 +152,201 @@ module _ {o ℓ} {C : Precategory o ℓ} where
...
+
+    id-kp : is-kernel-pair C id id f
+    id-kp .square = refl
+    id-kp .universal {p₁' = p₁'} _ = p₁'
+    id-kp .p₁∘universal = idl _
+    id-kp .p₂∘universal {p = p} = idl _ ∙ f-monic _ _ p
+    id-kp .unique p q = sym (idl _) ∙ p
+```
+
+Conversely, if $(id, id)$ is the kernel pair of $f$, then $f$ is monic.
```
Comment (ncfavier) [inline src/Cat/Diagram/Pullback.lagda.md:269]: `\id`

File: src/Cat/Diagram/Pullback.lagda.md [hunk 7]
```diff
@@ -152,6 +152,201 @@ module _ {o ℓ} {C : Precategory o ℓ} where
...
+    → is-monic f
+  id-kernel-pair→monic {f = f} id-kp g h p =
+    g                ≡˘⟨ p₁∘universal ⟩
+    id ∘ universal p ≡⟨ p₂∘universal ⟩
+    h                ∎
+    where open is-pullback id-kp
+```
+
+We can strengthen this result by noticing that if $(p, p)$ is the kernel
+pair of $f$ for some $P : \cC, p : P \to X$, then $(id, id)$ is also
```
Comment (ncfavier) [inline src/Cat/Diagram/Pullback.lagda.md:287]: `\id`

**Review by ncfavier — COMMENTED** [review]
`\id`

**Review by ncfavier — APPROVED** [review]

**Pull request merged**